src/core/bond_exist.cpp

```diff
@@ -0,0 +1,22 @@
+#include"bond_exist.hpp"
+#include"interaction_data.hpp"
+#include"bond.hpp"
+bool make_bond_exist_bond_class(int bond_type){
```

Comment:
Wouldn't a simple bonds_ia.resize(bond_type+1) suffice here? It is perfectly find to initialize the new bond to nullptr, it should not be accessed w/o being initialized anyway. I think a more sensible solution would be something like

```c++
void set_bond_by_type(int type, std::unique_ptr<Bond> && bond) {
  if(type >= bonds_ia.size()) {
    bonds_ia.resize(type + 1);
 }

 bonds_ia[type] = std::move(bond);
}
```